Source/Client/Windows/HostWindow.cs

```diff
@@ -29,30 +29,27 @@ public class HostWindow : Window
 
         private float height;
 
+        private ServerSettings settings;
+
         public HostWindow(SaveFile file = null, bool withSimulation = false)
         {
             closeOnAccept = false;
             doCloseX = true;
 
+            settings = MultiplayerMod.settings.serverSettings;
+
             this.withSimulation = withSimulation;
             this.file = file;
             settings.gameName = file?.gameName ?? Multiplayer.session?.gameName ?? $"{Multiplayer.username}'s game";
 
             var localAddr = MpUtil.GetLocalIpAddress() ?? "127.0.0.1";
             settings.lanAddress = localAddr;
-            addressBuffer = localAddr;
```

Comment:
addressBuffer was replaced with direct references to settings.directAddress in the effort to get settings to persist, but in doing so, we never default the address to the value returned by MpUtil.GetLocalIpAddress(). Instead this is now defaulting to 127.0.0.1 every time (see default value of serverAddress in MultiplayerMod.cs) This is a change in behavior, but it barely matters, since the first time the player sets the address it should get saved and recalled every time after.
I would be fine with this change.

Reply:
The behaviour of serverAddress was not changed at all. Not only it was 127.0.0.1 by default but it's also unrelated to the host window at all. What changed is the default address that server binds to which is now 0.0.0.0:30502.

Reply:
Ah yes, indeed. The default of 0.0.0.0 should work fine. (Binds to all local ips)